src/uiHelpers_nanox.c

```diff
@@ -0,0 +1,113 @@
+#include <os_io_seproxyhal.h>
+#if defined(TARGET_NANOX)
+#include <string.h>
+
+#include "uiHelpers.h"
+#include "ux.h"
+#include "assert.h"
+#include "io.h"
+#include "utils.h"
+#include "securityPolicy.h"
+
+// Helper macro for better astyle formatting of UX_FLOW definitions
+#define LINES(...) { __VA_ARGS__ }
+
+void scrolling_confirm()
+{
+    // TODO(ppershing): exception handling
+    TRY_CATCH_UI({
+        scrollingState_t* ctx = scrollingState;
```

Comment:
this should assert on init magic. We should probably extract the check to 
`assert_uiScrolling_magic()`

Reply:
done